mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -581,22 +581,23 @@ class Word2VecModel private[spark] (
     blas.sgemv(
       "T", vectorSize, numWords, alpha, wordVectors, vectorSize, fVector, 1, beta, cosineVec, 1)
 
-    val cosVec = cosineVec.map(_.toDouble)
-    var ind = 0
-    while (ind < numWords) {
-      val norm = wordVecNorms(ind)
+    var i = 0
+    while (i < numWords) {
+      val norm = wordVecNorms(i)
       if (norm == 0.0) {
-        cosVec(ind) = 0.0
+        cosineVec(i) = 0
```

Comment:
I'd still write 0.0f for clarity but it's no big deal. I guess we can write `norm == 0.0f` too